File: practice/algorithms/implementation/between_two_sets/BetweenTwoSets.js
'use strict';

const range = (start, end) => {
    const ret = [];
    for(let i = start; i <= end; i++) {
        ret.push(i);
    }
    return ret;
}

const processData = input => {
    const lines = input.split('\n');
    const a = lines[1].split(' ').map(i => parseInt(i));
    const b = lines[2].split(' ').map(i => parseInt(i));
    const aMax = Math.max(...a);
    const bMin = Math.min(...b);
    const result = range(aMax, bMin).filter(num => a.every(i => num % i === 0) && b.every(j => j % num === 0));
    console.log(result.length);
};

process.stdin.resume();
process.stdin.setEncoding("ascii");

let _input = "";
process.stdin.on("data", input => _input += input);
process.stdin.on("end", () => processData(_input));

File: practice/algorithms/implementation/between_two_sets/between_two_sets.rb
(n, m) = gets.split.map(&:to_i)
a = gets.split.map(&:to_i)
b = gets.split.map(&:to_i)
a_max = a.max
b_min = b.min
all_numbers = (a_max..b_min).to_a
result = all_numbers.select do |num|
  factor_of_a = a.all? { |i| num % i == 0 }
  factor_of_b = b.all? { |j| j % num == 0 }
  factor_of_a && factor_of_b
end

puts result.size